### PR TITLE
fix(HMS-3731): workaround for cloudwatch type field

### PR DESCRIPTION
**Pull request by avisiedo**
This workaround would fix the cloudwatch situation, so
the type field is filled with `"coudwatch"` string when
the cloudwatch values are merged on the service configuration.

When https://github.com/RedHatInsights/clowder/pull/627 PR
is merged, we can revert the first commit of this PR.

### CODE REVIEW

File: internal/config/config.go
Comment (cryptomilk) [inline internal/config/config.go:295]: Maybe also check if `len(clowderConfig.Logging.Cloudwatch.SecretAccessKey) > 0`?

Comment (tiran) [reply internal/config/config.go:295]: How about we only check for secret access keys? That way the code will still work when https://github.com/RedHatInsights/clowder/pull/627 has been merged.

Comment (avisiedo) [reply internal/config/config.go]: IMHO I don't know if another logging providers will coexists at some moment in the future; the guard `Type == ""` avoid to overwrite if some value exists at that moment, the check `len(clowderConfig.Logging.Cloudwatch.SecretAccessKey) > 0` will additionally set the value when we have some information for cloudwatch.

The current change accomplish:
- avoid overwrite value if some content exists.
- avoid to set the value if maybe cloudwatch object exists but the contents are empty.
- this let to set `"cloudwatch"` value meanwhile the PR is merged and will let the code works after merging it.

Comment (cryptomilk) [reply internal/config/config.go]: I think the workaround is good as it is right now. In future you might have cloudwatch secrets but logging.type configured to something else and want to react on this.

**Review by cryptomilk — APPROVED** [review]
LGTM

File: internal/config/config.go
Comment (tiran) [inline internal/config/config.go:297]: Let's future proof the check (untested code)

```suggestion
		// TODO Remove check for empty string after clowder has been fixed.
		// See: https://github.com/RedHatInsights/clowder/pull/627
		if (clowderConfig.Logging.Type == "" || clowderConfig.Logging.Type == "cloudwatch") && len(clowderConfig.Logging.Cloudwatch.SecretAccessKey) > 0 {
			v.Set("logging.type", "cloudwatch")
		}
```

Comment (cryptomilk) [reply internal/config/config.go:297]: Why set the logging type to cloudwatch if it already set? You might want to check the code with more context.

**Comment by cryptomilk** [comment]
I've talked to the author of the clowder fix (psav). They discovered two years ago, that they always set logging.type to "" instead of "cloudwatch". However a lot of projects already check for "" as the cloudwatch case so they didn't commit to fix it. He said all projects would have to change.

However it is a bug and should be fixed, but in the meantime a lot more projects use cloudwatch and the number of projects implementing a workaround increases. So the question is if them will ever fix it or not. If they don't fix it, they should check all the documentations and remove logging.type = "cloudwatch".

I think that the current implementation is the best workaround whatever they decide :-)

**Review by tiran — COMMENTED** [review]
LGTM, please rebase and squash all commits into a single commit.

**Comment by avisiedo** [comment]
My apologize, I realized about squash the commits when already pressed the merge button.

**Pull request merged**
